### ISSUE TITLE
Difference between weight decay and L2 regularization.

### ISSUE THREAD

**Issue by IljaAvadiev** [World4AI/World4AI#56]
The regularization chapter does not make any distinctions between L2 and weight decay and treats them as alternatives. It is possible, that the implementations are different though.

See the below link

https://benihime91.github.io/blog/machinelearning/deeplearning/python3.x/tensorflow2.x/2020/10/08/adamW.html
